stdlib/2and3/mmap.pyi

```diff
@@ -71,6 +70,9 @@ if sys.version_info >= (3,):
         def __setitem__(self, index: int, object: int) -> None: ...
         @overload
         def __setitem__(self, index: slice, object: bytes) -> None: ...
+        # Doesn't actually exist, but the object is actually iterable because it has __getitem__ and
+        # __len__, so we claim that there is also an __iter__ to help type checkers.
```

Comment:
Sorry for a late response. Yes, this is something that current version of PEP 544 will not solve. There is a similar problem with ``Reversible`` protocol ("new style"/``__reversed__`` vs "old style"/``__getitem__&__len__``). ``zope.interfaces`` has a concept of adaptors that aims to solve such problems, also there is rejected [PEP 246](https://www.python.org/dev/peps/pep-0246/), but I think adapters is quite an advanced concept. Also mypy and PEP 484 support unions and generic aliases, and if I understand everything correctly, they will solve this kind of problems, for example:
```python
class Iterable(Protocol[T]):
    def __iter__(self) -> Iterator[T]: ...
class OldIterable(Sized, Protocol[T]):
    def __getitem__(self, item: int) -> T: ...

CompatIterable = Union[Iterable[T], OldIterable[T]]

class A:
    def __iter__(self) -> Iterator[str]: ...
class B:
    def __len__(self) -> int: ...
    def __getitem__(self, item: int) -> str: ...

def iterate(it: CompatIterable[str]) -> None:
    ...

iterate(A())  # OK
iterate(B())  # OK
```
and similarly for ``Reversible``.

I therefore propose _not_ to include adaptation in PEP 544, probably we should add this to postponed/rejected section.